File: app/src/main/java/com/example/casaportemporada/Activity/Autenticacao/ResetPassword.java
package com.example.casaportemporada.Activity.Autenticacao;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageButton;
import android.widget.TextView;
import android.widget.Toast;

import com.example.casaportemporada.Helper.FirebaseHelper;
import com.example.casaportemporada.R;

public class ResetPassword extends AppCompatActivity {
    private ImageButton btn_back_to_login;
    private TextView text_message_resetPassword;

    private EditText text_resetpassword_email;

    private Button btn_resetPassword;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reset_password);

        startElements();
        eventClick();
    }


    private void startElements(){
        btn_back_to_login = findViewById(R.id.btn_resetPassword_back_login);
        text_message_resetPassword = findViewById(R.id.text_message_resetPassword);
        text_resetpassword_email = findViewById(R.id.text_resetpassword_email);
        btn_resetPassword = findViewById(R.id.btn_resetPaasword);
    }

    private void eventClick(){
        btn_back_to_login.setOnClickListener(view ->{
            startActivity(new Intent(this, LoginActivity.class));
        });
        btn_resetPassword.setOnClickListener(view ->{
            validateEmail();
        });

    }

    private void validateEmail(){
        String email = text_resetpassword_email.getText().toString();
        if (!email.isEmpty()){
            resetPassword(email);
        }else {
            text_resetpassword_email.requestFocus();
            text_resetpassword_email.setError("Digite seu email");
        }
    }

    public void resetPassword(String email){
        FirebaseHelper.getAuth().sendPasswordResetEmail(
            email
        ).addOnCompleteListener(task -> {
            if(task.isSuccessful()){
            text_message_resetPassword.setText("Acabamos de te enviar um email com um link para recupear sua senha.");
            }else {
                String error = task.getException().getMessage();
                Toast.makeText(this, error, Toast.LENGTH_SHORT).show();
            }
        });
    }

}

File: app/src/main/java/com/example/casaportemporada/activity/autenticacao/ResetPassword.kt
package com.example.casaportemporada.activity.autenticacao

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ImageButton
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.casaportemporada.Helper.FirebaseHelper
import com.example.casaportemporada.R
import com.google.android.gms.tasks.Task

class ResetPassword : AppCompatActivity() {
    private var btn_back_to_login: ImageButton? = null
    private var text_message_resetPassword: TextView? = null
    private var text_resetpassword_email: EditText? = null
    private var btn_resetPassword: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_reset_password)
        startElements()
        eventClick()
    }

    private fun startElements() {
        btn_back_to_login = findViewById(R.id.btn_resetPassword_back_login)
        text_message_resetPassword = findViewById(R.id.text_message_resetPassword)
        text_resetpassword_email = findViewById(R.id.text_resetpassword_email)
        btn_resetPassword = findViewById(R.id.btn_resetPaasword)
    }

    private fun eventClick() {
        btn_back_to_login!!.setOnClickListener { view: View? ->
            startActivity(
                Intent(
                    this,
                    LoginActivity::class.java
                )
            )
        }
        btn_resetPassword!!.setOnClickListener { view: View? -> validateEmail() }
    }

    private fun validateEmail() {
        val email = text_resetpassword_email!!.text.toString()
        if (!email.isEmpty()) {
            resetPassword(email)
        } else {
            text_resetpassword_email!!.requestFocus()
            text_resetpassword_email!!.error = "Digite seu email"
        }
    }

    fun resetPassword(email: String?) {
        FirebaseHelper.getAuth().sendPasswordResetEmail(
            email!!
        ).addOnCompleteListener { task: Task<Void?> ->
            if (task.isSuccessful) {
                text_message_resetPassword!!.text =
                    "Acabamos de te enviar um email com um link para recupear sua senha."
            } else {
                val error = task.exception!!.message
                Toast.makeText(this, error, Toast.LENGTH_SHORT).show()
            }
        }
    }
}
